chromium_src/chrome/browser/ui/browser_navigator.cc

```diff
@@ -5,5 +5,30 @@
 #include "brave/browser/renderer_host/brave_navigation_ui_data.h"
 
 #define ChromeNavigationUIData BraveNavigationUIData
+#define Navigate Navigate_ChromiumImpl
+#define IsURLAllowedInIncognito IsURLAllowedInIncognito_ChromiumImpl
 #include "../../../../chrome/browser/ui/browser_navigator.cc"
 #undef ChromeNavigationUIData
+#undef Navigate
+#undef IsURLAllowedInIncognito
+
+void Navigate(NavigateParams* params) {
```

Comment:
this is exactly what we don't want to do: try to find every place we need to hook in to map the url. We need to map it once and only once in a place that takes effect before any security checks

Reply:
I tried to test this place as a replacement of current mapping not for another hooking with current one. So I inserted `NOTREACHED()` there.

Reply:
Of course, I'll still studying navigation flow to find better place :)

Reply:
yep, just need to remember that there are many, many different paths and most of the are initiated by the renderer